src/Core/WinSWCore/LogAppenders.cs

```diff
@@ -470,21 +475,78 @@ private void ZipFiles(string directory, string fileExtension, string zipFileBase
                     string sourceFileName = Path.GetFileName(path);
                     string zipFilePattern = fileInfo.LastAccessTimeUtc.ToString(ZipDateFormat);
                     string zipFilePath = Path.Combine(directory, $"{zipFileBaseName}.{zipFilePattern}.zip");
-                    ZipOneFile(path, sourceFileName, zipFilePath);
-
-                    File.Delete(path);
+                    bool zipResult = ZipOneFile(path, sourceFileName, zipFilePath);
+                    if (zipResult)
+                    {
+                        File.Delete(path);
+                    }
                 }
             }
             catch (Exception e)
             {
                 EventLogger.LogEvent($"Failed to Zip files. Error {e.Message}");
             }
+
+            if(!(ZipDaysToKeep is null) && ZipDaysToKeep != 0)
+            {
+                try
+                {
+                    //Get all the zipfiles that might match the naming pattern
+                    List<string> ZipFiles = new List<string>(Directory.GetFiles(directory, zipFileBaseName + ".*.zip"));
+                    //Now delete the ones we don't want to keep, skipping any that don't match the naming pattern
+                    foreach ( string zipFilePath in ZipFiles)
+                    {
+                        if (File.Exists(zipFilePath))
+                        {
+                            //ZipDateFormat could contain literal characters, including the '.', so this gets a bit tricky
+                            string fileNameOnly = Path.GetFileNameWithoutExtension(zipFilePath);
+                            string[] parts = fileNameOnly.Split('.');
+                            string datePart;
+                            if(parts.Length > 2)
+                            {
+                                datePart = parts[1];
+                                for(int index = 2; index < parts.Length; index++)
+                                {
+                                    datePart += "." + parts[index];
+                                }
+                            }
+                            else
+                            {
+                                datePart = parts[parts.Length - 1];
+                            }
+                            //EventLogger.LogEvent($"{zipFilePath}: datePart = {datePart}");
+                            if (DateTime.TryParseExact(datePart, ZipDateFormat, System.Globalization.CultureInfo.InvariantCulture, System.Globalization.DateTimeStyles.None, out DateTime date))

```

Comment:
Do we use `InvariantCulture` when formatting `ZipDateFormat` to string?

Reply:
No, CurrentCulture is used when formatting the file name. TryParseExact call changed to match.